### PR TITLE
[HandshakeToFIRRTL] rewrite getBundleType method using a new getFIRRTLType method

### PR DESCRIPTION
**Pull request by hanchenye**
I found sometimes we may need to convert a standard type to its corresponding FIRRTL type, rather than an assembled bundle type containing data, valid, and ready. Therefore, I factored out a getFIRRTLType() method, and rewrote the getBundleType() accordingly.

### CODE REVIEW

**Review by mikeurbach — APPROVED** [review]
Looks like a nice refactor to me.

File: lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp [hunk 1]
```diff
@@ -32,10 +32,46 @@ using ValueVectorList = std::vector<ValueVector>;
...
+/// and none.
+static FIRRTLType getFIRRTLType(Type type) {
+  MLIRContext *context = type.getContext();
+  return TypeSwitch<Type, FIRRTLType>(type)
+      .Case<IntegerType>([&](IntegerType integerType) {
+        unsigned width = integerType.getWidth();
+
+        switch (integerType.getSignedness()) {
+        case IntegerType::Signed:
+          return FIRRTLType(SIntType::get(context, width));
```
Comment (lattner) [inline lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp:46]: I don't think you need the casts here, this should be enough:

`return SIntType::get(context, width);`

**Review by lattner — APPROVED** [review]
LGTM

**Comment by hanchenye** [comment]
Thanks for the review!

**Pull request merged**